### DISCUSSION TITLE
Structs (Anatomia)

### DISCUSSION THREAD

**Discussion by cmsouzaac** [balta-io/2801#113] [Perguntas e Respostas] [answered]
![image](https://user-images.githubusercontent.com/48097255/188534264-179f11c9-217a-4669-b5d7-38108ec69e8c.png)

Professor estudando a aula de `struct `como pode perceber, eu coloquei metodo `public` como **construtor/propriedade** e logo abaixo fazendo parte da anatomia, os metodos. Essa explicação está correta? estudo tambem JavaScript, e la toda função que possui o metodo "new" em sua composição é chamado de construtor. e aqui no c# o metodo new se encontra dentro da class estruct (ao qual eu crie).
![image](https://user-images.githubusercontent.com/48097255/188534773-8133ac63-fd84-4b06-a06f-6c147fd23271.png)

Poderia me explicar melhor essa estrutura e realmente o conceito do metodo `new`.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia @cmsoouza , como vai?

O JS trabalha um pouco diferente do C#, mas aqui também chamamos de construtor o método principal que não tem retorno e sempre é executado quando o objeto é construído.

```csharp
public class Teste
{
    public Teste() {} // Construtor
}
```

**Reply by cmsouzaac** [reply]
Neste caso entao todas estruturas que possuem essa estrutura a cima e com o metodo `new `sao construtores? pois as estrutura que coloquei como exemplo nao possuem o metodo `new`.

**Reply by andrebaltieri** [reply]
@cmsoouza sim! Sempre que você ver um método com o mesmo nome da classe e sem retorno (Nem void), significa que ele é o método construtor.
